File: ion/src/com/koushikdutta/ion/loader/AsyncHttpRequestFactory.java
package com.koushikdutta.ion.loader;

import android.net.Uri;

import com.koushikdutta.async.http.AsyncHttpRequest;
import com.koushikdutta.async.http.libcore.RawHeaders;

/**
 * Created by koush on 7/15/13.
 */
public interface AsyncHttpRequestFactory {
    public AsyncHttpRequest createAsyncHttpRequest(Uri uri, String method, RawHeaders headers, Object cookie);
}

File: ion/src/com/koushikdutta/ion/loader/AsyncHttpRequestFactory.kt
package com.koushikdutta.ion.loader

import com.koushikdutta.scratch.http.AsyncHttpMessageContent
import com.koushikdutta.scratch.http.AsyncHttpRequest
import com.koushikdutta.scratch.http.Headers
import com.koushikdutta.scratch.uri.URI

/**
 * Created by koush on 7/15/13.
 */
interface AsyncHttpRequestFactory {
    fun createAsyncHttpRequest(uri: URI, method: String, headers: Headers?, body: AsyncHttpMessageContent?): AsyncHttpRequest
}